### PR TITLE
radioboat: init at 0.2.1

### PR DESCRIPTION
**Pull request by zendo**
###### Description of changes

Radioboat is a terminal web radio client, built with simplicity in mind 

https://github.com/slashformotion/radioboat

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by azahi — CHANGES_REQUESTED** [review]
I've noticed an unhealthy increase of CPU usage (up to 100% for me) after I close `mpv`. It stays like this until I forcefully `kill -9` the main process because the interface itself becomes unresponsive. I'm pretty sure that's an upstream issue though. I think there's some kind of a loop that queries the state of `mpv` but since that process no longer exists, it just keeps repeating that over and over again.

**Comment by zendo** [comment]
I don't have `mpv` problem, when quit `radioboat` also `mpv` will disappeared in background process.

**Comment by azahi** [comment]
Don't you want to patch the [version](https://github.com/slashformotion/radioboat/blob/10e8d1e5d80034487666d8d9d7d0f97e53ee0872/cmd/root.go#L24) out as well so the program will print out the correct one? We probably should also notify upstream that the version will always say `v0.0.1` regardless.

**Comment by zendo** [comment]
It should be fix someday, i don't want to patch.

**Comment by azahi** [comment]
"Someday" doesn't include the version you are trying to package right now.

**Comment by superherointj** [comment]
Pending:
* To fix version, as it is hardcoded, use `substituteInPlace` to monkey patch it.
* Add test for version & simple execution (--help)

**Comment by zendo** [comment]
It doesn't affect anything, would be fix in upstream, and auto-update by `nix-update-script`.

**Comment by superherointj** [comment]
> It doesn't affect anything

There are guidelines. And checking for version is part of the process.

> would be fix in upstream, and auto-update by `nix-update-script`.

Realize all updates are reviewed. Auto-update only generates a PR for review. The process is same. Whenever upstream updates the application, the package will be updated accordingly. Please try to understand.

**Comment by slashformotion** [comment]
Updated the app, you will need to set the linker flags manually if you build the app yourself. See the makefile and https://goenning.net/2017/01/25/adding-custom-data-go-binaries-compile-time/ to learn more. 

If you need more info, please ask in a discution [here](https://github.com/slashformotion/radioboat/discussions)

File: pkgs/applications/audio/radioboat/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "slashformotion";
+    repo = "radioboat";
+    rev = "v${version}";
+    sha256 = "sha256-63O2nqT4HIK3Xo8VbNxVs0fwCmh/FUbok4P+nhpe3wI=";
+  };
+
+  vendorSha256 = "sha256-X3KiqaiOQYQBfVckh50C+4oxIVN6gXyNuQtBwGvjdFQ=";
+
+  ldflags = [ "-s" "-w" ];
```
Comment (slashformotion) [inline pkgs/applications/audio/radioboat/default.nix:16]: This should be changed to include -X flags

**Comment by slashformotion** [comment]
@zendo Please tell ping me when your "nixos" package is ready, I can include a link in the readme of Radioboat.

**Comment by zendo** [comment]
> @zendo Please tell ping me when your "nixos" package is ready, I can include a link in the readme of Radioboat.

Thanks for your work, it's a very nice radio app.

**Comment by superherointj** [comment]
@zendo radioboat has shell completion. Please add it. Something like:

```
  nativeBuildInputs = [ installShellFiles ];

  postInstall = ''
    installShellCompletion --cmd radioboat \
         --bash <($out/bin/radioboat completion bash) \
         --fish <($out/bin/radioboat completion fish) \
         --zsh <($out/bin/radioboat completion zsh)
  '';
```

Also add a test for version or simple test --help.
```
{ /.../, testers, radioboat }:

/.../
  passthru.tests.version = testers.testVersion {
    package = radioboat;
    command = "radioboat version";
    version = version;
  };
```

To test: `nix build .#radioboat.passthru.tests.version`

**Comment by zendo** [comment]
https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages

**Comment by superherointj** [comment]
> https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages

What do you mean specifically?

**Review by superherointj — APPROVED** [review]

**Comment by zendo** [comment]
> @zendo Please tell ping me when your "nixos" package is ready, I can include a link in the readme of Radioboat.

Finally we have `Radioboat` in `NixOS unstable`.
https://search.nixos.org/packages?channel=unstable&query=radioboat

**Comment by slashformotion** [comment]
@zendo v0.2.2 is out correcting 1 bug and adding 1 new feature.

**Comment by superherointj** [comment]
> @zendo v0.2.2 is out correcting 1 bug and adding 1 new feature.

The bot will open a new PR on this automatically. And maintainers will be notified. In case you want to be notified too, you could add yourself as a maintainer. Also, consider giving Nix a try, it is quite nice. You can install Nix on any linux distribution or even Mac. Or there is also [NixOS](https://nixos.org) which is a full Linux distribution.

**Pull request merged**